src/server/api/lib/campaign.ts

```diff
@@ -140,134 +140,147 @@ export interface CopyCampaignOptions {
-  const { db, campaignId, userId } = options;
+  const { db, campaignId, userId, quantity = 1 } = options;
 
   const result = await db.primary.transaction(async (trx) => {
-    // Copy campaign
-    const {
-      rows: [newCampaign]
-    } = await trx.raw<QueryResult<CampaignRecord>>(
-      `
-        insert into campaign (
-          organization_id,
-          title,
-          description,
-          is_approved,
-          is_started,
-          is_archived,
-          due_by,
-          logo_image_url,
-          intro_html,
-          primary_color,
-          texting_hours_start,
-          texting_hours_end,
-          timezone,
-          creator_id,
-          is_autoassign_enabled,
-          limit_assignment_to_teams,
-          replies_stale_after_minutes,
-          external_system_id
-        )
-        select
-          organization_id,
-          'COPY - ' || title,
-          description,
-          false as is_approved,
-          false as is_started,
-          false as is_archived,
-          due_by,
-          logo_image_url,
-          intro_html,
-          primary_color,
-          texting_hours_start,
-          texting_hours_end,
-          timezone,
-          ? as creator_id,
-          false as is_autoassign_enabled,
-          limit_assignment_to_teams,
-          replies_stale_after_minutes,
-          external_system_id
-        from campaign
-        where id = ?
-        returning *
-      `,
-      [userId, campaignId]
-    );
+    const cloneSingle = async (count: number) => {
+      // Copy campaign
+      const {
+        rows: [newCampaign]
+      } = await trx.raw<QueryResult<CampaignRecord>>(
+        `
+          insert into campaign (
+            organization_id,
+            title,
+            description,
+            is_approved,
+            is_started,
+            is_archived,
+            due_by,
+            logo_image_url,
+            intro_html,
+            primary_color,
+            texting_hours_start,
+            texting_hours_end,
+            timezone,
+            creator_id,
+            is_autoassign_enabled,
+            limit_assignment_to_teams,
+            replies_stale_after_minutes,
+            external_system_id
+          )
+          select
+            organization_id,
+            (case
+              when is_template then replace(concat('COPY - ', title), '#', ?::text)
+              else 'COPY - ' || title
+            end),
+            description,
+            false as is_approved,
+            false as is_started,
+            false as is_archived,
+            due_by,
+            logo_image_url,
+            intro_html,
+            primary_color,
+            texting_hours_start,
+            texting_hours_end,
+            timezone,
+            ? as creator_id,
+            false as is_autoassign_enabled,
+            limit_assignment_to_teams,
+            replies_stale_after_minutes,
+            external_system_id
+          from all_campaign
```

Comment:
is there a reason to now select from `all_campaign` instead of `campaign` like previously?

Reply:
ooh, just noticed it's introduced in the resolver fix